<a href="https://colab.research.google.com/github/khatriprajwol/Data-Visualization-/blob/main/SQL%20Essential.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## SQL Essential

In this project, I will be using the same data which I have used in the previous project. The data is about the 'austin_bikeshare.bikeshare_trips'.

In [ ]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

This code will is used to authorize the permission to use the data which included several process. 

In [4]:
%%bigquery --project data-base-1-338717
SELECT 
  COUNT(*) as total_rows
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

total_rows
0     1424786

It looks like there is total of 1424786 rows in the data set which is a lot. Lets have a look at the data and find out the attributes.

In [5]:
%%bigquery --project data-base-1-338717
SELECT *
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
LIMIT 15

trip_id  ... duration_minutes
0   9900289692  ...               39
1   9900285987  ...               31
2   9900285989  ...               31
3   9900285991  ...               30
4   9900286140  ...               19
5   9900286143  ...               17
6   9900286171  ...                6
7   9900286214  ...                0
8   9900286540  ...               12
9     13575843  ...               47
10  9900287291  ...                0
11  9900287293  ...                6
12  9900287294  ...                0
13  9900287414  ...                3
14  9900287861  ...               59

[15 rows x 9 columns]

This input gives me only 15 rows even though it has total of 1424786 rows. That is really helpful to over view in a small table which beautifully presents the information. 

In [6]:
%%bigquery --project data-base-1-338717 df1
SELECT 
  (duration_minutes), (start_station_name)
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

In [7]:
df1[1:30]

duration_minutes               start_station_name
1                 59                 Waller & 6th St.
2                 58                 Waller & 6th St.
3                 82          Toomey Rd @ South Lamar
4                 76                 Waller & 6th St.
5                 75                 Waller & 6th St.
6                 74                 Waller & 6th St.
7                 73                 Waller & 6th St.
8                 44                 Zilker Park West
9                 56                 Zilker Park West
10               453  State Capitol @ 14th & Colorado
11               246                 Zilker Park West
12                49          Toomey Rd @ South Lamar
13                53                 Zilker Park West
14               121          Toomey Rd @ South Lamar
15               122          Toomey Rd @ South Lamar
16               155                 Zilker Park West
17                48                     Lavaca & 6th
18                88                Rainey @ River St
19                49                     Lavaca & 6th
20                87                Rainey @ River St
21                87                Rainey @ River St
22                73                     Nueces @ 3rd
23                86                Rainey @ River St
24                71                     Nueces @ 3rd
25                71                 Zilker Park West
26                56                 Zilker Park West
27               148               11th & San Jacinto
28               173         ACC - West & 12th Street
29                54                 Zilker Park West

In [ ]:
df1.plot(kind='bar')

Exception ignored in: <function TransformNode.set_children.<locals>.<lambda> at 0x7fc29916aa70>
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/matplotlib/transforms.py", line 178, in <lambda>
    target=child._parents: target.pop(sid))
KeyboardInterrupt


KeyboardInterrupt: ignored

Wow! The graph gave me more than i was expecting. It gave me average time too. 

In [ ]:
%%bigquery --project data-base-1-338717
SELECT SUM(duration_minutes) as Total_Duration_Minutes
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

This value here is the total time from the starting point.

In [ ]:
%%bigquery --project data-base-1-338717 df2
SELECT AVG(duration_minutes) as avg_time, start_station_name
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
GROUP BY start_station_name
Order by avg_time

In [ ]:
df2[1:30]

In [ ]:
df2.plot(kind='bar')

In [ ]:
%%bigquery --project data-base-1-338717
SELECT AVG(duration_minutes) as duration_minutes
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips` as table

30.870428 is the average *duration_minutes* for the trip. We can also tell from the graph with out doing this calculation. 

In [ ]:
%%bigquery --project data-base-1-338717 df3
SELECT COUNT(start_station_name) as No_Trip, start_station_name
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
GROUP BY start_station_name
Order by No_Trip

In [ ]:
df3[1:30]

In [ ]:
df3.plot(kind='bar')

This graph looks kind of messy however, it is very imformative than the table how the data is behaving. 

Also, I just wanted to try to make a graph for df1 and df2. However, looking at the graph it is easy to tell the avarage time was 30.87 and the graph gave me the average time for the start point. It was interesting how informative graph can be. 